source/loop.cpp

```diff
@@ -350,6 +409,64 @@ double Loop::CalculateRadiativeLoss(double temperature)
 	return chi * std::pow( 10.0, (alpha*log_temperature) );
 }
 
+double Loop::CalculateRadiativeLoss(double temperature, double density)
+{
+    double abundance_array[] = {1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0};  // Discrete values available for abundance factors
+    int array_length = sizeof(abundance_array) / sizeof(abundance_array[0]);
+    double abundance_factor = CalculateAbundanceFactor(density);
+
+    // Find the nearest value of AF to the values in our discrete list, then return that value
+    int abundance_index = find_closest(abundance_factor, abundance_array, array_length);
+    
+    double log10_density_array[] = {8.0, 8.69897, 9.0, 9.69897, 10.0, 10.69897, 11.0}; // Discrete values available for density
```

Comment:
Same comment as above. I really don't like hardcoding these values like this and would prefer storing the values of the density in the files themselves. That way, if the density array ever changes, we don't have to update it in multiple places.